File: pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascal = [[1], [1, 1]]
        if numRows <= 2:
            return pascal[:numRows]
        else:
            for i in range(2, numRows):
                tmp = [1]
                for j in range(1, len(pascal[i - 1])):
                    l = pascal[i - 1][j - 1] + pascal[i - 1][j];
                    tmp.append(l)
                tmp.append(1)
                pascal.append(tmp)
        return pascal
            
        

File: pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> answer;
        for(int i = 1; i <= numRows; i++){
            vector<int> value(i);
            value[0] = 1;
            value[i - 1] = 1;
            for(int j = 1; j < i - 1; j++){
                value[j] = answer[i - 2][j - 1] + answer[i - 2][j];
            }
            answer.push_back(value);
        }
        return answer;
    }
};